Gruppe.A1.End-to-End_Encryption_mit_IPv6_und_integriertem_IPSec.ThushjandanPonnudurai.PascalBaumann.Termpaper.tex

```diff
@@ -111,19 +112,104 @@ \subsection{IPsec}
+	\item Blowfish (128, 192 or 256 bit)
+	\item CAST (128-bit)
+	\item Camellia (128, 192 or 256 bit)
+\end{itemize}
+AES-256 is commonly used in VPNs. DES is not secure anymore, because a DES key can be cracked in less than 24 hours. \parencite{Bollapragada2005}
+
+\subsubsection{Asymmetric}
+\label{sssec:asymmetric}
+Asymmetrical algorithms, also known as public key algorithm, use separate keys for encryption and decryption. The public key is used for the encryption and can be made public. The private key is used for decryption and must be kept secret. The public key and private key are mathematically related, but it is not possible to find out the private key from the public key. 
+
+RSA, DSA and Elliptic Curve (ECC) are some asymmetrical algorithms. \parencite{Bollapragada2005}
+
+\subsubsection{Diffie-Hellman (DH)}
+\label{sssec:diffie-hellman}
+With Diffie-Hellman algorithm (DH) two devices can establish a shared secret over an unsecure network. It was published by Whitfield Diffie and Martin Hellman in 1976. Related to VPN, it is used in IKE Phase 1 and 2 for key exchange. 
+
+Diffie Hellman group are used to specify the strength of the session key used in the DH key exchange process. Higher DH group numbers are more secure, but require additional processing resources to compute the session key. \parencite{JajishThomas}
+\begin{itemize}
+	\item \textbf{DH group 1}: 768-bit modulus
+	\item \textbf{DH group 2}: 1024-bit modulus
+	\item \textbf{DH group 5}: 1536-bit modulus
+	\item \textbf{DH group 14}: 2048-bit modulus
+	\item \textbf{DH group 15}: 3072-bit modulus
+	\item \textbf{DH group 19}: 256-bit elliptic curve group
+	\item \textbf{DH group 20}: 384-bit elliptic curve group
+	\item \textbf{DH group 21}: 521-bit elliptic curve group
+	\item \textbf{DH group 24}: 2048-bit modulus with 256 bit Prime Order subgroup
+\end{itemize}
+
+At the moment it is strongly recommended to use at least \textbf{DH group 14}. If an encryption or authentication algorithm with a 256-bit key or higher is used, then it is recommended to use DH group 21 or 24. \parencite{StrongSwan}
+
+\subsubsection{Hashing Algorithm}
+\label{sssec:hashing_algo}
+"Hash algorithms are also called \textit{digital fingerprinting algorithms}. They are irreversible functions that provide a fixed-size hash based on various inputs. Irreversibility and collision resistance are necessary attributes for successful hash functions." \parencite{Kampanakis2015}
+Message-Digest-Algorithm 5 (MD5), Secure Hash Algorithm 1 (SHA-1) or SHA-2 are some examples of hashing algorithms. MD5 and SHA-1 are \textbf{insecure} and should not be used. 
+
+A Keyed-Hash Message Authentication Code (HMAC) is a sort of message authentication codes (MAC) that uses a secret key and a cryptographic hash function. It is used for 
+\textbf{integrity verification}.
+
+There are existing several HMAC algorithms with various length of key strengths. Some of them are listed in the following list:
```

Comment:
Currently, several HMAC algorithm with various key strengths exist.
Or what do you mean with "length of key strengths", strength through key length?